script.js

```diff
@@ -1,4 +1,350 @@
+    } else {
+      totalHandValue = totalHandValue + currCard.rank;
+    }
+    index = index + 1;
+  }
+
+  // Reset index for ace counter
+  index = 0;
+  // Loop for the number of aces found and only deduct 10 from total hand value
+  // when totalHandValue is more than 21.
+  while (index < aceCounter) {
+    if (totalHandValue > 21) {
+      totalHandValue = totalHandValue - 10;
+    }
+    index = index + 1;
+  }
+
+  return totalHandValue;
+};
+
+// Function that displays the player and dealers hand in a message
+var displayPlayerAndDealerHands = function (playerHandArray, dealerHandArray) {
+  var playerMessage = "Player hand:<br>";
+  var index = 0;
+  while (index < playerHandArray.length) {
+    playerMessage =
+      playerMessage +
+      "- " +
+      playerHandArray[index].name +
+      " of " +
+      playerHandArray[index].suit +
+      "<br>";
+    index = index + 1;
+  }
+
+  index = 0;
+  var dealerMessage = "Dealer hand:<br>";
+  while (index < dealerHandArray.length) {
+    dealerMessage =
+      dealerMessage +
+      "- " +
+      dealerHandArray[index].name +
+      " of " +
+      dealerHandArray[index].suit +
+      "<br>";
+    index = index + 1;
+  }
+
+  return playerMessage + "<br>" + dealerMessage;
+};
+
+// Function that displays the total hand values of the player and the dealer in a message
+var displayHandTotalValues = function (playerHandValue, dealerHandValue) {
+  var totalHandValueMessage =
+    "<br>Player total hand value: " +
+    playerHandValue +
+    "<br>Dealer total hand value: " +
+    dealerHandValue;
+  return totalHandValueMessage;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var outputMessage = "";
+
+  // FIRST CLICK
+  if (currentGameMode == GAME_START) {
+    // create a deck of cards
+    gameDeck = createNewDeck();
+
+    // deal 2 cards to player and dealer
+    playerHand.push(gameDeck.pop());
+    playerHand.push(gameDeck.pop());
+    dealerHand.push(gameDeck.pop());
+    dealerHand.push(gameDeck.pop());
+
+    // check player and dealer cards
+    console.log("Player Hand ==>");
+    console.log(playerHand);
+    console.log("Dealer Hand ==>");
+    console.log(dealerHand);
+
+    // update gameMode
+    currentGameMode = GAME_CARDS_DRAWN;
+
+    // reassign output message
+    outputMessage =
+      "Everyone has been dealt a card. Click button to calculate cards!";
+
+    // return message
+    return outputMessage;
+  }
+
+  // SECOND CLICK
+  if (currentGameMode == GAME_CARDS_DRAWN) {
+    // check for blackjack
+    var playerHasBlackJack = checkForBlackJack(playerHand);
+    var dealerHasBlackJack = checkForBlackJack(dealerHand);
+
+    console.log("Does Player have Black Jack? ==>", playerHasBlackJack);
+    console.log("Does Dealer have Black Jack? ==>", dealerHasBlackJack);
+
+    // Condition when either player or dealer has black jack
+    if (playerHasBlackJack == true || dealerHasBlackJack == true) {
```

Comment:
Similarly, you would almost never need to do == true since the variable itself should be a boolean

```js
if (playerHasBlackJack || dealerHasBlackJack)
```